### PR TITLE
Migrate to GH Actions

### PR DESCRIPTION
**Pull request by rxmarbles**
Move to GH actions still need to add secrets and then we can merge

### CODE REVIEW

File: .github/workflows/cicd.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      - name: Install dependencies
+        run: npm ci
+      - name: Test
```
Comment (gergelyke) [inline .github/workflows/cicd.yml:21]: can you please add the TS type checks and the linter as well?

**Review by gergelyke — APPROVED** [review]
looks great, thanks @rxmarbles

File: .github/workflows/cicd.yml (hunk 1 above)
Comment (rxmarbles) [reply .github/workflows/cicd.yml:21]: @gergelyke the main `test` function runs the typings. Did you want that separated into its own line item? If so we can remove it from the main `npm test` command.

Comment (gergelyke-dd) [reply .github/workflows/cicd.yml:21]: my bad - disregard please :)

Comment (rxmarbles) [reply .github/workflows/cicd.yml:21]: No worries 😄 I did add a job for linting to run after test so we are all good there

**Pull request merged**